Amplify/Categories/Auth/Result/AuthSignInResult.swift

```diff
@@ -13,18 +13,15 @@ public struct AuthSignInResult {
     ///
     /// When this value is false, it means that there are more steps to follow for the signIn flow. Check `nextStep`
     /// to understand the next flow. If `isSignedIn` is true, signIn flow has been completed.
-    public let isSignedIn: Bool
+    public var isSignedIn: Bool { nextStep == .done }
 
     /// Shows the next step required to complete the signIn flow.
     ///
-    /// This value will be nil if `isSignedIn` is true.
-    public var nextStep: AuthSignInStep?
+    public var nextStep: AuthSignInStep
```

Comment:
nit: this is no longer a `nextStep`, let's make sure to cover this during API review